bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/src/main/java/com/google/cloud/bigtable/hbase1_x/BigtableBenchmark.java

```diff
@@ -0,0 +1,211 @@
+  static final int ROWS_PER_BATCH = 10000;
+
+  @Param("")
+  private String projectId;
+
+  @Param("")
+  private String instanceId;
+
+  @Param({
+    "cellsPerRow/10/cellSize/100",
+    "cellsPerRow/1/cellSize/1024",
+    "cellsPerRow/100/cellSize/1"
+  })
+  private String benchmarkConfig;
+
+  @Param({"true", "false"})
+  private boolean useGcj;
+
+  @Param("false")
+  private boolean useBatch;
+
+  private Connection connection;
+
+  private BenchmarkJobConfig jobConfig;
+
+  @Setup
+  public void setUp() throws IOException {
+    connection = BenchmarkDataUtils.createConnection(projectId, instanceId, useBatch, useGcj);
+    jobConfig = new BenchmarkJobConfig(benchmarkConfig);
+
+    createAndFillTable(connection, jobConfig);
+    createTableToWrite(connection, TableName.valueOf(BENCHMARK_WRITE_TABLE));
+  }
+
+  @Benchmark
+  public void pointRead(Blackhole blackhole) throws IOException {
+    Table table = connection.getTable(jobConfig.tableName);
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+    Result result = table.get(new Get(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, 1))));
+
+    for (Cell cell : result.listCells()) {
+      blackhole.consume(cell);
+    }
+  }
+
+  @Benchmark
+  public void bulkScans(Blackhole blackhole) throws IOException {
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+
+    Scan scan = new Scan().setMaxVersions(1).setRowPrefixFilter(Bytes.toBytes(rowPrefix));
+    ResultScanner resScanner = connection.getTable(jobConfig.tableName).getScanner(scan);
+
+    for (Result result = resScanner.next(); result != null; result = resScanner.next()) {
+
+      for (Cell cell : result.rawCells()) {
+        blackhole.consume(cell);
+      }
+    }
+  }
+
+  // TODO(rahulkql): yet to be fully tested.
+  @Benchmark
+  public void pointWrite() throws IOException {
+    Table table = connection.getTable(TableName.valueOf(BENCHMARK_WRITE_TABLE));
+
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+    Put put = new Put(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, 100)));
+    long currentTimestamp = System.currentTimeMillis();
+
+    for (int cellInd = 0; cellInd < jobConfig.cellsPerRow; cellInd++) {
+      put.addColumn(
+          COL_FAMILY,
+          Bytes.toBytes("qualifier-" + String.format(SUFFIX_FMT, cellInd)),
+          currentTimestamp,
+          getRandomBytes(jobConfig.cellSize));
+    }
+
+    table.put(put);
+  }
+
+  // TODO(rahulkql): yet to be fully tested.
+  @Benchmark
+  public void bulkWrite() throws IOException {
+    try (BufferedMutator bufferedMutator =
+        connection.getBufferedMutator(TableName.valueOf(BENCHMARK_WRITE_TABLE))) {
+
+      String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+      byte[] cellValue = getRandomBytes(jobConfig.cellSize);
+      ImmutableList.Builder<Put> putBuilder = ImmutableList.builderWithExpectedSize(ROWS_PER_BATCH);
```

Comment:
Why do we need an intermediate list? can't we just add directly to the buffered mutator?

Reply:
Actually i was flushing after 10K ross that's why I created an intermediate builder. Now removed it as we are only sending 100 rows and JMH is iterating it multiple times.